File: apps/backend/Ecommerce/Ecommerce.Domain/Entities/Banner.cs
namespace Ecommerce.Domain.Entities
{
    public class Banner : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public string ButtonText { get; set; } = string.Empty;
        public string ButtonLink { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
    }
}

File: apps/frontend/ecommerce-client/types/banner.ts
export interface Banner {
    id: string
    title: string
    description: string
    imageUrl: string
    buttonText: string
    buttonLink: string
    isActive: boolean
    order: number
    createdAt: string
    updatedAt: string
}

export interface BannersResponse {
    banners: Banner[]
    total: number
    page: number
    limit: number
    totalPages: number
}

export interface BannerFilters {
    isActive?: boolean
    page?: number
    limit?: number
    search?: string
}
